lib/filter.c

```diff
@@ -171,22 +204,37 @@ static int filter_match_cisco(struct filter *mfilter, struct prefix *p)
 }
 
 /* If filter match to the prefix then return 1. */
-static int filter_match_zebra(struct filter *mfilter, struct prefix *p)
+static int filter_match_zebra(struct filter *mfilter, void *obj)
```

Comment:
It being `struct ethaddr` is the issue, it should be `struct prefix_eth` instead, then it can be passed as `struct prefix`